code/game/objects/items/devices/scanners.dm

```diff
@@ -296,41 +307,139 @@ REAGENT SCANNER
 				dat += "\tBlood Level normal: [blood_percent]% [blood_volume]cl. Type: [blood_type]\n"
 		// Show pulse
 		dat += "\tPulse: <font color='[H.pulse == PULSE_THREADY || H.pulse == PULSE_NONE ? "red" : ""]'>[H.get_pulse(GETPULSE_TOOL)] bpm.</font>\n"
-		if((H.stat == DEAD && !H.client) || HUSK in H.mutations)
-			unrevivable = 1
+		if(H.stat == DEAD)
+			is_dead = TRUE
+			//check to see if the target is revivable
+			if(!H.is_revivable())
+				unrevivable = 1
 		if(!unrevivable)
+			//Chems that conflict with others:
+			var/synaptizine_amount = reagents_in_body["synaptizine"]
+			var/hyperzine_amount = reagents_in_body["hyperzine"]
+			var/paracetamol_amount = reagents_in_body["paracetamol"]
+			//Recurring chems:
+			var/peridaxon = ""
+			var/tricordrazine = ""
+			//The actual medical advice summary:
 			var/advice = ""
+			//We start checks for ailments here:
+			if(is_dead)
+				var/death_message = ""
+				//Check for whether there's an appropriate ghost
+				if(H.client)
+					//Calculate revival status/time left
+					var/revive_timer = round((H.timeofdeath + H.revive_grace_period - world.time) * 0.1)
+					death_message = "Reduce total injury value to sub-200, then administer defibrillator to unarmoured chest before brain death in [revive_timer] seconds."
```

Comment:
Okay, this is a major change that needs to be made clear in the PR description and discussed.